### PR TITLE
feat: add state properties to worker

### PR DESCRIPTION
**Pull request by metcoder95**

### CODE REVIEW

File: src/worker_pool/index.ts [hunk 1]
```diff
@@ -124,6 +136,7 @@ export class WorkerInfo extends AsynchronouslyCreatedResource {
     }
 
     processPendingMessages () {
+      if (this.terminating || this.destroyed) return;
```
Comment (metcoder95) [inline src/worker_pool/index.ts]:
```suggestion
      if (this.destroyed) return;
```

File: test/workers.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { resolve } from 'node:path';
+
+import { test } from 'tap';
+
+import Piscina from '../dist';
+
+test('workers are marked as detroyed if destroyed', async t => {
```
Comment (ronag) [inline test/workers.ts]:
```suggestion
test('workers are marked as destroyed if destroyed', async t => {
```

File: src/worker_pool/index.ts [hunk 3]
```diff
@@ -156,13 +169,25 @@ export class WorkerInfo extends AsynchronouslyCreatedResource {
     get interface (): PiscinaWorker {
       const parent = this;
       return {
-        id: this.worker.threadId.toString(),
-        currentUsage: this.currentUsage(),
-        isRunningAbortableTask: this.isRunningAbortableTask(),
+        get id () {
+          return parent.worker.threadId.toString();
```
Comment (ronag) [inline src/worker_pool/index.ts]: Why to string?

**Review by ronag — APPROVED** [review]

File: src/worker_pool/index.ts (hunk 3 above)
Comment (metcoder95) [reply src/worker_pool/index.ts]: Its a number by default, was wondering if shall we keep it like this or better parse it as string

Comment (ronag) [reply src/worker_pool/index.ts]: I would suggest to keep it as a number. More efficient to use than a string.

**Pull request merged**